File: Airbnb-Backend/Airbnb.Core/Entities/Listing.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Airbnb.Core.Entities
{
    public class Listing : BaseEntity
    {
        public string Title { get; set; }

        public string Description { get; set; }

        public string PropertyType { get; set; }

        public string RoomType { get; set; }

        public int Capacity { get; set; }
        
        public int Bedrooms { get; set; }

        public int Bathrooms { get; set; }

        public decimal Price_Per_Night { get; set; }

        public decimal Service_Fee { get; set; } = 0;

        public string Address_Line1 { get; set; }

        public string? Address_Line2 { get; set; }

        public string City { get; set; }

        public string State { get; set; }

        public string Country { get; set; }

        public string PostalCode { get; set; }

        public decimal? Latitude { get; set; }

        public decimal? Longitude { get; set; }

        public DateTime Created_At { get; set; }

        public DateTime Updated_At { get; set; }

        public int Min_Nights { get; set; } = 1;

        public int? Max_Nights { get; set; }

        public string? Cancellation_Policy { get; set; }

        public decimal? Average_Rating { get; set; }

        public int Review_Count { get; set; } = 0;

        public bool IsActive { get; set; } = true;


        public int Host_Id { get; set; }
        public User Host { get; set; }







    }
}

File: Airbnb-Frontend/src/app/core/models/Listing.ts
// export interface Listing {
//   id: string;
//   host: {
//     id: string;
//     firstName: string;
//     lastName: string;
//     profilePictureUrl: string;
//     bio: string;
//     isHost: boolean;
//     isVerified: boolean;
//   };
//   title: string;
//   description: string;
//   propertyTypeId: number;
//   roomTypeId: number;
//   capacity: number;
//   bedrooms: number;
//   bathrooms: number;
//   pricePerNight: number;
//   serviceFee: number;
//   addressLine1: string;
//   addressLine2: string;
//   city: string;
//   state: string;
//   country: string;
//   postalCode: string;
//   latitude: number;
//   longitude: number;
//   instantBooking: boolean;
//   createdAt: string;
//   updatedAt: string;
//   minNights: number;
//   maxNights: number;
//   cancellationPolicyId: number;
//   averageRating: number;
//   reviewCount: number;
//   isActive: boolean;
//   currencyId: number;
//   imageUrls: string[];
//   previewImageUrl: string;
//   amenities: {
//     id: string;
//     name: string;
//     categoryId: string;
//     icon: string;
//     createdAt: string;
//   }[];
//   reviews: {
//     id: string;
//     bookingId: string;
//     reviewer: {
//       id: string;
//       firstName: string;
//       lastName: string;
//       profilePictureUrl: string;
//       bio: string;
//       isHost: boolean;
//       isVerified: boolean;
//     };
//     hostId: string;
//     listingId: string;
//     comment: string;
//     hostReply: string;
//     hostReplyDate: string;
//     createdAt: string;
//     updatedAt: string;
//     rating: number;
//     cleanlinessRating: number;
//     accuracyRating: number;
//     communicationRating: number;
//     locationRating: number;
//     checkInRating: number;
//     valueRating: number;
//   }[];

// }


export interface Host {
  id: string;
  firstName: string;
  lastName: string;
  profilePictureUrl: string;
  bio: string;
  isHost: boolean;
  isVerified: boolean;
}

export interface CancellationPolicy {
  id: number;
  name: string;
  description: string;
  fullRefundDays: number | null;
  partialRefundDays: number | null;
  partialRefundPercentage: number | null;
}

export interface Listing {
  id: string;
  host: Host;
  title: string;
  description: string;
  propertyTypeId: number;
  roomTypeId: number;
  capacity: number;
  bedrooms: number;
  bathrooms: number;
  pricePerNight: number;
  serviceFee: number;
  addressLine1: string;
  addressLine2: string;
  city: string;
  state: string;
  country: string;
  postalCode: string;
  latitude: number;
  longitude: number;
  instantBooking: boolean;
  createdAt: string; // You can use `Date` if you're parsing it
  updatedAt: string;
  minNights: number;
  maxNights: number;
  cancellationPolicy: CancellationPolicy;
  averageRating: number;
  reviewCount: number;
  isActive: boolean;
  currencyId: number;
  verificationStatusId: number;
  imageUrls: string[];
  previewImageUrl: string;
  amenities: any[]; // You can replace `any` with a specific `Amenity` type if available
  reviews: any[];   // Same here for `Review`
}
